pandas/core/arrays/datetimelike.py

```diff
@@ -430,14 +430,6 @@ def astype(self, dtype, copy: bool = True):
         elif is_integer_dtype(dtype):
             # we deliberately ignore int32 vs. int64 here.
             # See https://github.com/pandas-dev/pandas/issues/24381 for more.
```

Comment:
im OK with un-deprecating for the time being for i8.  any objection to maintaining the deprecation for integer dtypes other than i8?

Reply:
Other integer dtypes should IMO _certainly_ use astype instead of view? (eg `view(int32)` doesn't make much sense since the original dtype is 64-bit)

Reply:
I'm saying if the user does `arr.astype(np.uint32)` it doesn't make sense to give them `arr.view(np.int64)`.  i.e. we should keep the deprecation for dtypes other than np.int64.

Reply:
And what I say is that it makes even less sense to ask them to do `.view(np.int32)` instead of giving them `.view(np.int64)` ;) (which will actually raise). 

I understand that we are ignoring the bit-width here, and that is something we should fix. But I think that is unrelated to the "use view instead" deprecation (I also don't know if that is something we should deprecate first, or simply change).

Reply:
Actually, we might be ignoring bit-width here, but that gets catched elsewhere. Because astyping to something else as int64/uint64 actually already raises an error at the moment (see also the table in https://github.com/pandas-dev/pandas/issues/45034#issuecomment-1000998887)

At least, that gets catched on the Series level. As you mentioned in that issue, the cast is allowed (bit width is ignored) at the array level.

Reply:
> And what I say is that it makes even less sense to ask them to do .view(np.int32) instead of giving them .view(np.int64) ;) (which will actually raise).

huh?  If a user does `dta.astype(np.uint32)` im not advocating returning `dta.view(np.int32)`, im advocating raising (or at least deprecating now and raising in the future).

> Actually, we might be ignoring bit-width here, but that gets catched elsewhere. Because astyping to something else as int64/uint64 actually already raises an error at the moment (see also the table in #45034 (comment))

Yah we should try to handle these all symmetrically.

Reply:
> huh? If a user does dta.astype(np.uint32) im not advocating returning dta.view(np.int32), im advocating raising (or at least deprecating now and raising in the future).

I know that you are not advocating for that, but is _is_ what the current deprecation warning implies the user to do (as is says to use `view` instead of `astype`). 
You asked to "maintain the deprecation" for those cases, and what I am trying to say is that we should indeed fix those cases (in some way, with a different deprecation or breaking change (depending on the future behaviour), to be discussed), but IMO not by maintaining the current deprecation.